src/Html.php

```diff
@@ -1103,17 +1211,28 @@ public static function checkboxList(string $name, $selection = null, array $item
         $name = self::getArrayableName($name);
 
         if (is_iterable($selection)) {
-            $selection = array_map('strval', (array)$selection);
+            $selection = array_map('strval', is_array($selection) ? $selection : iterator_to_array($selection));
```

Comment:
Originally it meant that if a single value is selected then you can specify it without wrapping it into array explicitly. Now it changed its meaning.

Reply:
No... this is only for iterable values and originally this correctly work with `ArrayObject` and not correctly with custom iterable object. For example, this test fail originally:

```php
Html::checkboxList('test', new IterableObject(['value2']), $this->getDataItems(), [
	'item' => static function ($index, $label, $name, $checked, $value) {
		return $index . Html::label($label . ' ' . Html::checkbox($name, $checked, ['value' => $value]));
	},
	'tag' => false,
]);
```

Now  single value as selected also work good.